### DISCUSSION TITLE
Warning while installing Langchain:  antlr4_python3_runtime-4.11.1-py3.12.egg is deprecated

### DISCUSSION THREAD

**Discussion by agnibhu13** [antlr/antlr4#4497] [General]
Hi,

while installing Langchain, I got the following warning related to a

DEPRECATION: Loading egg at c:\users\deepu\appdata\local\programs\python\python312\lib\site-packages\antlr4_python3_runtime-4.11.1-py3.12.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330

Seems like the egg based distribution is deprecated in pip.

**Comment by ericvergnaud** [comment]
Hi, seems that's a Langchain issue. Have they upgraded to 4.13.1 ? (can't guarantee it addresses the issue but IIRC we did some change in packaging some time ago)
